File: USACO/Training Website/Chapter 1/Section 1.2 - Ad Hoc/ride.py
'''
ID: sunnyky1
LANG: PYTHON3
TASK: ride
'''

file_in = open("ride.in", mode='r')
comet_name = file_in.readline()
group_name = file_in.readline()
comet_total = 1
for char in comet_name[:-1]:
    comet_total *= (ord(char) - 64)
group_total = 1
for char in group_name[:-1]:
    group_total *= (ord(char) - 64)
if comet_total % 47 == group_total % 47:
    output = "GO"
else:
    output = "STAY"
file_out = open("ride.out", mode='w')
file_out.write(output + "\n")
file_out.close()

File: USACO/Training Website/Chapter 1/Section 1.2 - Ad Hoc/ride.java
/*
ID: sunnyky1
LANG: JAVA
TASK: ride
*/

import java.io.*;

public class ride {

	public static void main(String[] args) throws IOException {
		BufferedReader fin = new BufferedReader(new FileReader("ride.in"));
		PrintWriter fout = new PrintWriter(new BufferedWriter(new FileWriter("ride.out")));
		if (convert(fin.readLine()) % 47 == convert(fin.readLine()) % 47) {
			fout.println("GO");
		} else {
			fout.println("STAY");
		}
		fin.close();
		fout.close();
	}
	
	static int convert(String str) {
		byte[] bytes = str.getBytes();
		int num = 1;
		for (int i = 0; i < bytes.length; i++) {
			num *= (int) bytes[i] - 'A' + 1;
		}
		return num;
	}

}
